File: public/docs/_examples/dependency-injection/ts/app/hero.service.ts
// #docregion
import {Hero} from './hero';
import {HEROES} from './mock_heroes';

class HeroService {

  heroes: Hero[];

  constructor() {
    this.heroes = HEROES;
  }

  getHeroes() {
    return this.heroes;
  }
}
// #enddocregion

File: public/docs/_examples/toh-4/dart/lib/hero_service.dart
// #docplaster
// #docregion
// #docregion just-get-heroes
import 'dart:async';

import 'package:angular2/core.dart';

import 'hero.dart';
import 'mock_heroes.dart';

@Injectable()
class HeroService {
  // #docregion get-heroes
  Future<List<Hero>> getHeroes() async => mockHeroes;
  // #enddocregion get-heroes, just-get-heroes
  // #enddocregion
  // See the "Take it slow" appendix
  // #docregion get-heroes-slowly
  Future<List<Hero>> getHeroesSlowly() {
    return new Future.delayed(const Duration(seconds: 2), () => mockHeroes);
  }
  // #enddocregion get-heroes-slowly
  // #docregion
  // #docregion just-get-heroes
}

